app/models/section_offering.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2024, Schweizer Alpen-Club. This file is part of
+#  hitobito_sac_cas and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito_sac_cas
+
+# == Schema Information
+#
+# Table name: section_offerings
+#
+#  id         :bigint           not null, primary key
+#  title      :text(65535)      not null
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+
+class SectionOffering < ApplicationRecord
+  has_and_belongs_to_many :sections,
+    class_name: Group::Sektion.sti_name,
+    association_foreign_key: :group_id
+
+  translates :title, fallbacks_for_empty_translations: true
+
+  validates :title, presence: true
+  before_destroy :check_associated_sections
+
+  default_scope { includes(:translations) }
+
+  def to_s
+    title
+  end
+
+  private
+
+  # Callback to check if there are any associated sections
```

Comment:
This is needed, because HABTM relations do not support `dependent: :restrict_with_error`, see: https://github.com/rails/rails/issues/32291#issuecomment-374216313

Since we most likely dont want a active record model for this join table, we can't use HMT relations, so I decided to just validate it with a `before_destroy`

Reply:
For the termination reasons it is solved by rescueing from `ActiveRecord::InvalidForeignKey`

Reply:
TerminationReason is not used in a habtm relation, we need the callback to prevent the record from being deleted when there's still groups referencing it. ActiveRecord throws no error that we could catch